src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -581,7 +581,12 @@ private void ProcessPingFrame(FrameHeader frameHeader)
 
             // Send PING ACK
             // Don't wait for completion, which could happen asynchronously.
-            LogExceptions(SendPingAckAsync(_incomingBuffer.ActiveMemory.Slice(0, FrameHeader.PingLength)));
+            // We discard the the frame from the incoming buffer, possibly before
+            // SendPingAckAsync is completed, so we need to save our own copy of the data.
+            // The data is not an integer in the RFC, but this will avoid an array allocation.
+            long pingContent = BinaryPrimitives.ReadInt64BigEndian(_incomingBuffer.ActiveSpan.Slice(0, FrameHeader.PingLength));
```

Comment:
If anything, move this line to above the "Send PING ACK" comment above, since that comment is specifically commenting on the call to SendPingAckAsync